hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClusterListenerSupport.java

```diff
@@ -200,8 +198,7 @@ private boolean connect(Set<InetSocketAddress> triedAddresses) throws Exception
                 fireConnectionEvent(LifecycleEvent.LifecycleState.CLIENT_CONNECTED);
                 return true;
             } catch (Exception e) {
-                Level level = e instanceof AuthenticationException ? Level.WARNING : Level.FINEST;
-                logger.log(level, "Exception during initial connection to " + inetSocketAddress, e);
+                logger.warning("Exception during initial connection to " + inetSocketAddress + ", exception " + e);
```

Comment:
It is. Actually currently if Authentication is configured wrong. It is printed twice :
```
WARNING: hz.client_0 [PrometheusUATx] [3.9-SNAPSHOT] ClientConnection{alive=false, connectionId=1, channel=NioChannel{/127.0.0.1:59831->/127.0.0.1:5701}, remoteEndpoint=null, lastReadTime=2017-05-29 15:11:51.743, lastWriteTime=2017-05-29 15:11:51.723, closedTime=2017-05-29 15:11:51.753, lastHeartbeatRequested=never, lastHeartbeatReceived=never, connected server version=null} lost. Reason: com.hazelcast.client.AuthenticationException[Invalid credentials! Principal: null]
May 29, 2017 3:11:51 PM com.hazelcast.client.spi.impl.ClusterListenerSupport
WARNING: hz.client_0 [PrometheusUATx] [3.9-SNAPSHOT] Exception during initial connection to /127.0.0.1:5701
com.hazelcast.client.AuthenticationException: Invalid credentials! Principal: null
	at com.hazelcast.client.connection.nio.ClientConnectionManagerImpl$1.onResponse(ClientConnectionManagerImpl.java:576)
```

In ssl, it is reported that we are missing SSL exceptions here. Not sure how it is missed.
There can be other exceptions other than connection get here. Adding membership listener could fail for example. 
I don't think  printing exception twice is a problem here. The other cases should be printed